cosmwasm/register-its-token.js

```diff
@@ -0,0 +1,128 @@
+    constructor() {
+        this.tokenIndex = -1;
+        this.tokenIds = Object.keys(tokenInfo);
+        this.incrementTokenIndex();
+        this.chainIndex = -1;
+    }
+
+    tokenId() {
+        return this.tokenIds[this.tokenIndex];
+    }
+
+    token() {
+        return tokenInfo[this.tokenId()];
+    }
+
+    incrementTokenIndex() {
+        if(this.tokenIndex >= this.tokenIds.length - 1) return false;
+        this.tokenIndex++;
+        this.chainNames = Object.keys(this.token()).slice(0, -1);
+        this.chainIndex = 0;
+        return true;
+    }
+
+    get() {
+        return this.token()[this.chainName()];
+    }
+
+    chainName() {
+        return this.chainNames[this.chainIndex];
+    }
+
+    rpc() {
+        const chainName = this.chainName();
+        return env === 'mainnet' || env === 'testnet' ? RPCs.axelar_bridge_evm.find((chain) => chain.name.toLowerCase() === chainName).rpc_addr : info.chains[chainName].rpc;
+    }
+
+    async getNext() {
+        const previous = this.get();
+        if (previous && previous.supply) delete previous.supply;
+        fs.writeFileSync(`./axelar-chains-config/info/tokens-${env}.json`, JSON.stringify(tokenInfo, null, 2));                 
+        while (true) {
+            this.chainIndex++;
+            if (this.chainIndex >= this.chainNames.length) {
+                if(!this.incrementTokenIndex()) return false;
+            }
+            const chainName = this.chainName()
+            if(chainName === this.token().originChain.origin_chain) continue;
+            const current = this.get();
+            if (!current.registered) {
+                if (current.track) try {
+                    const provider = getDefaultProvider(this.rpc());
+                    const token = new Contract(current.tokenAddress, IInterchainToken.abi, provider);
+                    current.supply = await token.totalSupply();
+                } catch (e) {
+                    printError('Failed to query token supply for', current.tokenAddress);
+                }
+                console.log(`Chain Progress: ${this.chainIndex + 1}/${this.chainNames.length} | Token Progress: ${this.tokenIndex + 1}/${this.tokenIds.length}`)
+                return true;
+            }
+        }
+    }
+}
+
+async function registerToken(tokenIterator) {
+    const config = loadConfig(env);
+
+    initContractConfig(config, {contractName: "InterchainTokenService"});
+
+    const wallet = await prepareWallet({mnemonic});
+    const client = await prepareClient(config, wallet);
+    const supply = tokenIterator.get().supply;
+    const supplyParam = supply ? {"tracked": String(supply)} : "untracked";
+    const msg = { "register_p2p_token_instance": {
+        "chain": tokenIterator.chainName(),
+        "token_id": tokenIterator.tokenId().slice(2),
+        "origin_chain": tokenIterator.token().originChain,
+        "decimals": tokenIterator.get().decimals,
+        "supply": supplyParam,
+    } };
```

Comment:
add a dry run mode to make it easy to verify what the cmd will do before running it for sanity